lib/protein/adapters/amqp_adapter/server.ex

```diff
@@ -45,9 +45,11 @@ defmodule Protein.AMQPAdapter.Server do
         chan
       :error ->
         reconnect_int = Utils.get_config(opts, :reconnect_interval, 5_000)
-        Logger.error(fn ->
-          "Connection to #{url} failed, reconnecting in #{reconnect_int}ms"
-        end)
+        error_message = "Connection to #{url} failed, reconnecting in #{reconnect_int}ms"
+        if remote_logger = Application.get_env(:protein, :remote_error_log) do
```

Comment:
Let's name the setting `custom_error_logger` - remote suggests a concept of networking and it should be logger not log because log is just a single entry not the logging target.